tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -2095,3 +2095,293 @@ def call(self, inputs, state):
+      logging.warn("%s: Using a concatenated state is slower and will soon be "
+                   "deprecated.  Use state_is_tuple=True.", self)
+    self._num_units = num_units
+    self._forget_bias = forget_bias
+    self._bias_start = bias_start
+    self._alpha_start = alpha_start
+    self._beta_start = beta_start
+    self._state_is_tuple = state_is_tuple
+    self._activation = activation
+    self._reuse = reuse
+
+  @property
+  def state_size(self):
+    return (rnn_cell_impl.LSTMStateTuple(self._num_units, self._num_units) 
+        if self._state_is_tuple else 2 * self._num_units)
+
+  @property
+  def output_size(self):
+    return self._num_units
+
+  def call(self, inputs, state):
+    """Run one step of Multiplicative Integration Long short-term memory(LSTM)."""
+    if self._state_is_tuple:
+      c, h = state
+    else:
+      c, h = array_ops.split(value=state, num_or_size_splits=2, axis=1)
+    concat = _multiplicative_integration([inputs, h],
+        [self._num_units, self._num_units, self._num_units, self._num_units],
+        True, bias_start=self._bias_start, alpha_start=self._alpha_start,
+        beta_start=self._beta_start)
+
+    # i = input_gate, j = new_input, f = forget_gate, o = output_gate
+    i, j, f, o = array_ops.split(value=concat, num_or_size_splits=4, axis=1)
+
+    new_c = (c * math_ops.sigmoid(f + self._forget_bias)
+                + math_ops.sigmoid(i) * self._activation(j))
+    new_h = self._activation(new_c) * math_ops.sigmoid(o)
+
+    if self._state_is_tuple:
+      new_state = rnn_cell_impl.LSTMStateTuple(new_c, new_h)
+    else:
+      new_state = array_ops.concat([new_c, new_h], 1)
+    return new_h, new_state
+
+
+def _multiplicative_integration(args, output_sizes, bias, bias_start=0.0, 
+    alpha_start=1.0, beta_start=1.0):
+  """Multiplicative Integration: alpha * args[0] * W[0] + beta1 * args[1] * W[1] 
+            + beta1 * args[2] * W[2],
+        where alpha, beta1, beta2 and W[i] are variables.
+
+  Args:
+    args: a list of 2D, batch x n, Tensors.
+    output_sizes: a list of second dimension of W[i], where list[i] is int.
+    bias: boolean, whether to add a bias term or not.
+    bias_start: float, starting value to initialize the bias; 0 by default.
+    alpha_start: float, starting value to initialize the alpha; 
+      1 by default.
+    beta_start: float, starting value to initialize the beta1 and beta2; 
+      1 by default.
+
+  Returns:
+    A 2D Tensor with shape [batch x output_size] equal 
+      to the Multiplicative Integration above.
+
+  Raises:
+    ValueError: if some of the arguments has unspecified or wrong shape.
+  """
+  if args is None or (nest.is_sequence(args) and not args):
+    raise ValueError("`args` must be specified")
+  if not nest.is_sequence(args):
+    raise ValueError("`args` must be list")
+  if len(args) != 2:
+    raise ValueError("`args` must contain 2 tensors")
+  if output_sizes is None or (nest.is_sequence(output_sizes) 
```

Comment:
`if not output_sizes`?

the only case you're allowing here is output_sizes == 0; and do you want to allow that?